src/main/java/org/skriptlang/skript/elements/fireworks/ExprFireworkFlownTime.java

```diff
@@ -0,0 +1,171 @@
+	 * Paper does not agree with Spigot's method naming, so they have Spigot's deprecated.
+	 */
+	static {
+		registerDefault(ExprFireworkFlownTime.class, Timespan.class, "[firework] ([detonate:max[imum]] life[time]|flown time|detonate:time until detonat(e|ion))", "fireworks");
+	}
+
+	private boolean detonate;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		detonate = parseResult.hasTag("detonate");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public Timespan convert(Firework firework) {
+		if (PAPER)
+			return Timespan.fromTicks_i(detonate ? firework.getTicksToDetonate() : firework.getTicksFlown());
+		return Timespan.fromTicks_i(detonate ? firework.getMaxLife() : firework.getLife());
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case DELETE:
+			case REMOVE:
+			case RESET:
+			case SET:
+				return CollectionUtils.array(Timespan.class);
+			case REMOVE_ALL:
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		long ticks = (mode == ChangeMode.DELETE || mode == ChangeMode.RESET) ? 0 : ((Timespan) delta[0]).getTicks_i();
+		switch (mode) {
+			case REMOVE:
+				ticks = -ticks;
+			case ADD:
+				for (Firework firework : getExpr().getArray(event)) {
+					long existing = convert(firework).getTicks_i();
+					if (detonate) {
+						if (PAPER) {
+							firework.setTicksToDetonate((int) Math.min(0, existing + ticks));
+						} else {
+							firework.setMaxLife((int) Math.min(0, existing + ticks));
+						}
+					} else {
+						if (PAPER) {
+							firework.setTicksFlown((int) Math.min(0, existing + ticks));
+						} else {
+							firework.setLife((int) Math.min(0, existing + ticks));
+						}
+					}
+				}
+				break;
+			case SET:
+				for (Firework firework : getExpr().getArray(event)) {
+					if (detonate) {
+						if (PAPER) {
+							firework.setTicksToDetonate((int) Math.min(0, ticks));
+						} else {
+							firework.setMaxLife((int) Math.min(0, ticks));
+						}
+					} else {
+						if (PAPER) {
+							firework.setTicksFlown((int) Math.min(0, ticks));
+						} else {
+							firework.setLife((int) Math.min(0, ticks));
+						}
+					}
+				}
+				break;
+			case DELETE:
+			case RESET:
+				for (Firework firework : getExpr().getArray(event)) {
+					if (detonate) {
+						if (PAPER) {
+							firework.setTicksToDetonate((int) 20 * firework.getFireworkMeta().getPower());
+						} else {
+							firework.setMaxLife((int) 20 * firework.getFireworkMeta().getPower());
+						}
+					} else {
+						if (PAPER) {
+							firework.setTicksFlown((int) 20 * firework.getFireworkMeta().getPower());
+						} else {
+							firework.setLife((int) 20 * firework.getFireworkMeta().getPower());
+						}
+					}
+				}
+				break;
```

Comment:
I think this is unintuitive. 
Delete should not be a valid changer, in my opinion, or if it is, it should set the value to 0.
Reset can't reliably figure out what the lifespan should be, since it doesn't know how long it's flown already.
For that reason, Reset probably also shouldn't be valid, or if it is, it should set remaining time to the original amount and the ticks flown to 0.